src/main/java/org/jabref/model/entry/BibEntry.java

```diff
@@ -112,6 +113,110 @@ public BibEntry(EntryType type) {
         return Optional.empty();
     }
 
+    /**
+     * Map an (empty) field of a BibEntry to a field of a cross-referenced entry.
+     *
+     * @param targetField field name of the BibEntry
+     * @param targetEntry type of the BibEntry
+     * @param sourceEntry type of the cross-referenced BibEntry
+     *
+     * @return the mapped field or null if there is no valid mapping available
+     */
+    private Field getSourceField(Field targetField, EntryType targetEntry, EntryType sourceEntry) {
```

Comment:
please use `Optional<Field>` as the return type (instead of returning null below).